### ISSUE TITLE
Docker images build breaks too often

### ISSUE THREAD

**Issue by eikek** [eikek/docspell#2374]
I'm not sure what to do about it yet, but I'm very tempted to just remove it. I don't use it anyways and it is taking too much of my time to maintain. The docker build breaks too often - it feels like every other week. It was fixed just a few days ago and it is failing again (probably because of some package dependency problem).

Maybe someone else wants to maintain it? It could also be a separate repo on the docspell org.

**Comment by LightTemplar** [comment]
What if you will do it for releases only? It's still a very convenient way to use software.

**Comment by eikek** [comment]
The problem are the releases. I don't care much about the snapshots, because it is in flux anyways. It can slow down releasing a lot. Often it is a boring task around chasing package problems (more often than not it's the ocrmypdf python tool). The images also need to be tested a bit. Unfortunately, this consumes much of my time, and I'm not using it myself. I think I'll move the docker things out in a separate repo. It can publish docker images and once it breaks maybe someone else can take care.

**Comment by LightTemplar** [comment]
I agree with this solution. Though, it would be interesting to know, if you have a statistics of Docspell installations. How big is the part, which goes onto docker?

**Comment by eikek** [comment]
Unfortunately, I have no idea. I _think_ it is a significant percentage, though. But I need to cut down time spent anyways.

**Comment by programmerq** [comment]
I'd like to see the official image stick around.

I'd recommend building off a stable alpine release instead of an alpine edge snapshot. The underlying edge repo is a rolling release, and things can change inadvertently.

Part of a release process would include manually bumping the major alpine version base to the latest supported and working alpine version.

That's my suggestion on this problem. I think it'd be a lot smoother and be less prone to breaking.

**Comment by eikek** [comment]
I did this in the past and it resulted in too much time spent hunting down alpine packages for that new version. So it only moves the annoying stuff to release time. And at the end I used to just stick with the version I had :)

**Comment by programmerq** [comment]
Sticking with a major version of alpine as opposed to using an edge release will yield fewer breakages.

Alternatively, an Ubuntu LTS base would allow you to have a relatively unchanged base image for years. If you're open to that, I'm happy to work on an Ubuntu variant and open a PR.

**Comment by eikek** [comment]
Ah thanks! I think I misunderstood. The idea was actually not to use "edge" but the latest *stable* release. So yes, this should be changed. I like your suggestion. If ubuntu is more stable, I would definitely prefer it, because I won't spend much time anymore on docker stuff myself :). As long as it builds I'm fine. When it breaks, I'll probably move it into a separate repo for others to maintain. 

One little advantage could be that it is easier to publish new images after a fix to the image only.

**Comment by pschichtel** [comment]
I guess this issue can be closed and given that the new image over at https://github.com/docspell/docker exists now.
